xchange-binance/src/main/java/org/knowm/xchange/binance/service/BinanceTradeService.java

```diff
@@ -99,30 +100,11 @@ public OpenOrders getOpenOrders(OpenOrdersParams params) throws IOException {
 
   @Override
   public String placeMarketOrder(MarketOrder mo) throws IOException {
-
-    return placeOrder(OrderType.MARKET, mo, null, null, null);
+    return Long.toString(placeOrder(OrderType.MARKET, mo, null, null, null).orderId);
   }
 
-  @Override
-  public String placeLimitOrder(LimitOrder lo) throws IOException {
-    TimeInForce tif = TimeInForce.GTC;
-    OrderType type;
-    if (lo.hasFlag(org.knowm.xchange.binance.dto.trade.BinanceOrderFlags.LIMIT_MAKER)) {
-      type = OrderType.LIMIT_MAKER;
-      tif = null;
-    } else {
-      type = OrderType.LIMIT;
-      Set<IOrderFlags> orderFlags = lo.getOrderFlags();
-      Iterator<IOrderFlags> orderFlagsIterator = orderFlags.iterator();
-
-      while (orderFlagsIterator.hasNext()) {
-        IOrderFlags orderFlag = orderFlagsIterator.next();
-        if (orderFlag instanceof TimeInForce) {
-          tif = (TimeInForce) orderFlag;
-        }
-      }
-    }
-    return placeOrder(type, lo, lo.getLimitPrice(), null, tif);
+  public BinanceNewOrder placeMarketOrder2(MarketOrder mo) throws IOException {
```

Comment:
remove this from here, just use the *Raw method in your application code if you want to get binances DTO

Reply:
Not useful, maybe it would be better to make public current private method placeOrder? Cause it contains some business logic and exception handling. What do you think?

Reply:
you can have public methods in *Raw clases
non *Raw classes should return / throw generic DTOs / exception
*Raw classes should return / throw exchange specific DTOs / exception